.github/workflows/wheel.yaml

```diff
@@ -1,39 +1,62 @@
 name: Create and publish Python package
 
 on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
   push:
+    tags:
+      - "v*"
+    branches:
+      - "main"
+  workflow_dispatch:
 
 jobs:
+  version:
+    uses: ./.github/workflows/versioning.yaml
+
   build:
-    runs-on:
-      - ubuntu-latest
+    runs-on: ubuntu-latest
+    needs: version
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
-        with:
-          fetch-tags: true
       - uses: actions/setup-python@v5
         with:
           python-version: '3.12'
+          cache: pip
+      - name: Install build tool
+        uses: pypa/hatch@install
```

Comment:
suggestion: the version of hatch lib can be precised 1.12.0 in case hatch is updated to avoid incompatibilities

Reply:
I prefer to let it as a flying version, and pin it if any error is raised. This workflow is ran often enough to quickly detect incompatibilities.